src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -1188,6 +1188,19 @@ private HttpResponseHeaders CreateResponseHeaders(bool appCompleted)
                 }
             }
 
+            // TODO allow customization of this.
+            if (ServerOptions.EnableAltSvc && _httpVersion < Http.HttpVersion.Http3) 
+            {
+                foreach (var option in ServerOptions.ListenOptions)
+                {
+                    if (option.Protocols == HttpProtocols.Http3)
+                    {
+                        responseHeaders.HeaderAltSvc = $"h3-25=\":{option.IPEndPoint.Port}\"; ma=84600";
```

Comment:
This is fine for now, but for Http/2 we should really implement the alt-service frame to avoid adding this header to all responses. File an issue?

Reply:
Sure!

Reply:
https://github.com/dotnet/aspnetcore/issues/18928